src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests.vb

```diff
@@ -5111,5 +5113,456 @@ DiagID1: 'C' is for evaluation purposes only and is subject to change or removal
+
+            Dim derivedSrc = <compilation>
+                                 <file name="a.vb">
+                                     <![CDATA[
+Class Derived
+    Inherits C
+End Class
+]]>
+                                 </file>
+                             </compilation>
+
+            Dim derivedComp = CreateCompilation(derivedSrc, references:={originalC.ToMetadataReference(), attrRef})
+            derivedComp.AssertNoDiagnostics()
+
+            Dim comp = CreateCompilation("", references:={derivedComp.ToMetadataReference(), retargetedC.ToMetadataReference(), attrRef})
+            derivedComp.AssertNoDiagnostics()
+
+            Dim derivedType = comp.GetTypeByMetadataName("Derived")
+            Assert.IsType(Of RetargetingNamedTypeSymbol)(derivedType)
+            Assert.IsType(Of RetargetingAssemblySymbol)(derivedType.ContainingAssembly)
+            Assert.Equal(ObsoleteAttributeKind.None, derivedType.ContainingAssembly.ObsoleteKind)
+        End Sub
+
+        <Fact>
+        Public Sub RetargetingModule_Experimental()
+            Dim attrComp = CreateCSharpCompilation(experimentalAttributeCSharpSrc)
+            Dim attrRef = attrComp.EmitToImageReference()
+
+            Dim retargetedCode = <compilation>
+                                     <file name="a.vb">
+                                         <![CDATA[
+Public Class C
+End Class
+]]>
+                                     </file>
+                                 </compilation>
+
+            Dim originalC = CreateCompilationWithIdentity(New AssemblyIdentity("Ret", New Version(1, 0, 0, 0), isRetargetable:=True), retargetedCode)
+            Dim retargetedC = CreateCompilationWithIdentity(New AssemblyIdentity("Ret", New Version(2, 0, 0, 0), isRetargetable:=True), retargetedCode)
+
+            Dim derivedSrc = <compilation>
+                                 <file name="a.vb">
+                                     <![CDATA[
+<Module: System.Diagnostics.CodeAnalysis.Experimental("DiagID1")>
+
+Class Derived
+    Inherits C
+End Class
+]]>
+                                 </file>
+                             </compilation>
+
+            Dim derivedComp = CreateCompilation(derivedSrc, references:={originalC.ToMetadataReference(), attrRef})
+            derivedComp.AssertNoDiagnostics()
+
+            Dim comp = CreateCompilation("", references:={derivedComp.ToMetadataReference(), retargetedC.ToMetadataReference(), attrRef})
+            derivedComp.AssertNoDiagnostics()
+
+            Dim derivedType = comp.GetTypeByMetadataName("Derived")
+            Assert.IsType(Of RetargetingNamedTypeSymbol)(derivedType)
+            Assert.IsType(Of RetargetingModuleSymbol)(derivedType.ContainingModule)
+            Assert.Equal(ObsoleteAttributeKind.Experimental, derivedType.ContainingModule.ObsoleteKind)
+        End Sub
+
+        <Fact>
+        Public Sub MissingAssemblyAndModule()
```

Comment:
>MissingAssemblyAndModule [](http://example.com/codeflow?start=19&length=24)

Is this testing the `<Experimental>` changes?

Reply:
Yes, we're exercising the change in `MissingAssemblySymbol` and `MissingModuleSymbol` by calling `ObsoleteKind` on those.